src/gainmap.c

```diff
@@ -126,6 +127,14 @@ avifResult avifRGBImageApplyGainMap(const avifRGBImage * baseImage,
 
     const uint32_t width = baseImage->width;
     const uint32_t height = baseImage->height;
+
+    const avifBool useBaseColorSpace = gainMap->metadata.useBaseColorSpace;
+    const avifColorPrimaries gainMapMathPrimaries =
+        (useBaseColorSpace || (gainMap->altColorPrimaries == AVIF_COLOR_PRIMARIES_UNSPECIFIED)) ? baseColorPrimaries
+                                                                                                : gainMap->altColorPrimaries;
```

Comment:
Because we are applying the gain map that is given to us, so we use the information in the gain map metadata.
avifChooseColorSpaceForGainMapMath() is used when creating a gain map, to find the best color space to do the math in. After that, the color space if fixed, and the results will be wrong if you use the other color space.